### ISSUE TITLE
deno publish warns about statically analyzable dynamic imports

### ISSUE THREAD

**Issue by kitsonk** [denoland/deno#22641]
Version: Deno 1.41.0

Currently zap/fast-check warns that certain dynamic imports are unanalyzable, when in fact they can be analyzed when the module specifier is set to a static string, for example the following warning is quite incorrect:

```
warning[unanalyzable-dynamic-import]: unable to analyze dynamic import
   --> /Users/kitsonk/github/oak/application.ts:600:46
    | 
600 |       const { NativeRequest } = await import("./http_server_native_request.ts");
    |                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the unanalyzable dynamic import

  info: after publishing this package, imports from the local import map / package.json do not work
  info: dynamic imports that can not be analyzed at publish time will not be rewritten automatically
  info: make sure the dynamic import is resolvable at runtime without an import map / package.json
```

Zap/fast-check should be able to analyze those, as they are statically resolvable at check time.

Related #22594

**Comment by bartlomieju** [comment]
I believe this is already fixed by https://github.com/denoland/deno/pull/22636 and will be released in v1.41.1

**Comment by dsherret** [comment]
Yeah, pretty sure this one is fixed in canary and doing a 1.41.1 release right now.

**Comment by kitsonk** [comment]
Yeah, confirm it works according to expectations from main now.
